### PR TITLE
Linting (clang-tidy)

### PR DESCRIPTION
**Pull request by egelja**

### CODE REVIEW

**Review by AlemSnyder — DISMISSED** [review]

**Comment by AlemSnyder** [comment]
invalid preprocessing directiveC/C++(clang-diagnostic-error)
on `#cmakedefine01 DEBUG()`

__float128 is not supported on this target
Clang does not support __float128, but we are not using clang to compile.

**Comment by MrAwesomeRocks** [comment]
> invalid preprocessing directiveC/C++(clang-diagnostic-error)
> on `#cmakedefine01 DEBUG()`

Yea it's right here, this is done by CMake

> __float128 is not supported on this target
> Clang does not support __float128, but we are not using clang to compile.

No this is a bug: https://github.com/microsoft/vscode-cpptools/issues/9198
And it just shows the error but still works.

**Comment by AlemSnyder** [comment]
Saying "I know it's buggy" is not a reason to use buggy code.

In any case clang-tity breaks, and ends up using 30% of my cpu for ~2 hours. I won't be using it.

File: src/world.hpp [hunk 1]
```diff
@@ -84,34 +85,40 @@ class World {
...
+    static std::vector<int>
+    get_grass_grad_data(const Json::Value& materials_json);
 
     /**
      * @brief Get the mesh using greedy meshing
      *
      * @return entity::Mesh the mesh generated
      */
-    inline std::vector<entity::Mesh> get_mesh_greedy() const {
+    [[nodiscard]] inline std::vector<entity::Mesh>
```
Comment (AlemSnyder) [inline src/world.hpp:116]: why are the [[nodiscard]] statements inconsistent?

**Pull request closed without merging**